src/main/java/edu/harvard/iq/dataverse/api/auth/CompoundAuthMechanism.java

```diff
@@ -19,9 +19,9 @@ public class CompoundAuthMechanism implements AuthMechanism {
     private final List<AuthMechanism> authMechanisms = new ArrayList<>();
 
     @Inject
-    public CompoundAuthMechanism(ApiKeyAuthMechanism apiKeyAuthMechanism, WorkflowKeyAuthMechanism workflowKeyAuthMechanism, SignedUrlAuthMechanism signedUrlAuthMechanism, SessionCookieAuthMechanism sessionCookieAuthMechanism, BearerTokenAuthMechanism bearerTokenAuthMechanism) {
```

Comment:
I was not aware of that. I do not have a strong opinion on that and I do not have a problem doing it as an filter. It is just less confusing to me since this approach implements only the IdentityStore for the (HttpAuthenticationMechanism)[ttps://hantsy.gitbook.io/java-ee-8-by-example/security/security-auth] and does not implement the HttpAuthenticationMechanism itself. I do not know any other way to implement the bearer tokens. Is there some reason why why do not use security annotations in the code? The only way I can think of implementing it as an AuthMechanism, is to write an AuthMechanism that injects the security context and delegates the work to the HttpAuthenticationMechanism that looks up the implemented identity stores and finds the `BearerTokenMechanism` identity store do its job. We could just have multiple identity store implementations instead and delegate the authentication to the HttpAuthenticationMechanism that we do not have to implement ourselves, like it happens in this PR. Was the previous implementation like that? Did it cause some problems? At the first glance it looks much less complicated to implement.